stdlib/src/builtin/string.mojo

```diff
@@ -2109,3 +2205,83 @@ fn _calc_format_buffer_size[type: DType]() -> Int:
         return 64 + 1
     else:
         return 128 + 1  # Add 1 for the terminator
+
+
+# ===----------------------------------------------------------------------===#
+# Format method structures
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct _FormatCurlyEntry:
```

Comment:
Q: Now that things are cleaning up, why do we need this struct at all? Looks like you never instantiate it, and it just has a single static method. Can we just have an ordinary function instead?

Reply:
Would love to, but we cannot return a list of tuples yet,  what else could we return ?

Reply:
Fair point, this is fine, thank you!